core/src/main/java/tech/tablesaw/io/fixed/FixedWidthReader.java

```diff
@@ -0,0 +1,620 @@
+
+        StringBuilder buf = new StringBuilder();
+        buf.append("ColumnType[] columnTypes = {");
+        buf.append('\n');
+
+        Column<?> typeCol = structure.column("Column Type");
+        Column<?> indxCol = structure.column("Index");
+        Column<?> nameCol = structure.column("Column Name");
+
+        // add the column headers
+        int typeColIndex = structure.columnIndex(typeCol);
+        int indxColIndex = structure.columnIndex(indxCol);
+        int nameColIndex = structure.columnIndex(nameCol);
+
+        int typeColWidth = typeCol.columnWidth();
+        int indxColWidth = indxCol.columnWidth();
+        int nameColWidth = nameCol.columnWidth();
+
+        final char padChar = ' ';
+        for (int r = 0; r < structure.rowCount(); r++) {
+            String cell = Strings.padEnd(structure.get(r, typeColIndex) + ",", typeColWidth, padChar);
+            buf.append(cell);
+            buf.append(" // ");
+
+            cell = Strings.padEnd(structure.getUnformatted(r, indxColIndex), indxColWidth, padChar);
+            buf.append(cell);
+            buf.append(' ');
+
+            cell = Strings.padEnd(structure.getUnformatted(r, nameColIndex), nameColWidth, padChar);
+            buf.append(cell);
+            buf.append(' ');
+
+            buf.append('\n');
+        }
+        buf.append("}");
+        buf.append('\n');
+        return buf.toString();
+    }
+
+    /**
+     * Reads column names from header, skipping any for which the type == SKIP
+     */
+    private String[] selectColumnNames(List<String> names, ColumnType[] types) {
+        List<String> header = new ArrayList<>();
+        for (int i = 0; i < types.length; i++) {
+            if (types[i] != SKIP) {
+                String name = names.get(i);
+                name = name.trim();
+                header.add(name);
+            }
+        }
+        String[] result = new String[header.size()];
+        return header.toArray(result);
+    }
+
+    /**
+     * Provides placeholder column names for when the file read has no header
+     */
+    private String[] makeColumnNames(ColumnType[] types) {
+        String[] header = new String[types.length];
+        for (int i = 0; i < types.length; i++) {
+            header[i] = "C" + i;
+        }
+        return header;
+    }
+
+    /**
+     * Estimates and returns the type for each column in the delimited text file {@code file}
+     * <p>
+     * The type is determined by checking a sample of the data in the file. Because only a sample of the data is
+     * checked,
+     * the types may be incorrect. If that is the case a Parse Exception will be thrown.
+     * <p>
+     * The method {@code printColumnTypes()} can be used to print a list of the detected columns that can be
+     * corrected and
+     * used to explicitly specify the correct column types.
+     */
+    public ColumnType[] detectColumnTypes(InputStream stream, FixedWidthReadOptions options) throws IOException {
```

Comment:
Can you use the `ColumnTypeDetector` class instead of duplicating its code here?